### PR TITLE
annotateWithTypeFromJSDoc: Use changes.insertTypeAnnotation instead of replaceNode

### LINKED ISSUE #22358: Codefix to annotate with type from jsdoc reformats entire function
**TypeScript Version:**  2.8.0-dev.20180302

**Code**

```ts
/** @param {number} x */
function f(x) {
    return x+1;
}
```

**Expected behavior:**

```ts
/** @param {number} x */
function f(x: number) {
    return x+1;
}
```

**Actual behavior:**

```ts
/** @param {number} x */
function f(x: number) {
    return x + 1;
}
```

Inferring the type from usage does not have this problem, so we could probably reuse some code from there.

### PR DESCRIPTION
**Pull request by andy-ms**
Fixes #22358

### CODE REVIEW

File: tests/cases/fourslash/annotateWithTypeFromJSDoc13.ts [hunk 1]
```diff
@@ -9,6 +9,6 @@ verify.codeFix({
     newFileContent:
 `class C {
     /** @return {number} */
-    get c(): number { return 12; }
+    get c(): number { return 12 }
```
Comment (amcasey) [inline tests/cases/fourslash/annotateWithTypeFromJSDoc13.ts]: Why did we drop the semicolon?

File: src/compiler/utilities.ts [hunk 2]
```diff
@@ -2839,49 +2839,38 @@ namespace ts {
...
      */
-    export function getEffectiveTypeAnnotationNode(node: Node, checkJSDoc?: boolean): TypeNode | undefined {
-        if (hasType(node)) {
-            return node.type;
-        }
-        if (checkJSDoc || isInJavaScriptFile(node)) {
-            return getJSDocType(node);
-        }
+    export function getEffectiveTypeAnnotationNode(node: Node): TypeNode | undefined {
+        return (node as HasType).type || (isInJavaScriptFile(node) ? getJSDocType(node) : undefined);
```
Comment (amcasey) [inline src/compiler/utilities.ts]: For my own understanding, if `node` is not a `HasType` but does have a `type` property with a defined value, we will return that (as we did before?)?

Comment (andy-ms) [reply src/compiler/utilities.ts]: Yes, the assumption is that only a `HasType` has a type.

File: tests/cases/fourslash/annotateWithTypeFromJSDoc13.ts (hunk 1 above)
Comment (andy-ms) [reply tests/cases/fourslash/annotateWithTypeFromJSDoc13.ts]: There's no semicolon in the original. Previously we would reformat the body, but now we just add a type annotation.

**Pull request merged**